frontend/src/metabase/admin/databases/components/widgets/EngineWidget/EngineWidget.tsx

```diff
@@ -0,0 +1,336 @@
+        />
+      ) : (
+        <EngineEmptyState isHosted={isHosted} />
+      )}
+      {!isSearching && hasMoreOptions && (
+        <EngineToggle
+          isExpanded={isExpanded}
+          onExpandedChange={setIsExpanded}
+        />
+      )}
+    </EngineSearchRoot>
+  );
+};
+
+interface EngineListProps {
+  rootId: string;
+  options: EngineOption[];
+  activeIndex?: number;
+  onOptionChange?: (value: string) => void;
+}
+
+const EngineList = ({
+  rootId,
+  options,
+  activeIndex,
+  onOptionChange,
+}: EngineListProps): JSX.Element => {
+  return (
+    <EngineListRoot role="listbox" id={getListBoxId(rootId)}>
+      {options.map((option, optionIndex) => (
+        <EngineCard
+          key={option.value}
+          rootId={rootId}
+          option={option}
+          isActive={optionIndex === activeIndex}
+          onOptionChange={onOptionChange}
+        />
+      ))}
+    </EngineListRoot>
+  );
+};
+
+export interface EngineCardProps {
+  rootId: string;
+  option: EngineOption;
+  isActive: boolean;
+  onOptionChange?: (value: string) => void;
+}
+
+const EngineCard = ({
+  rootId,
+  option,
+  isActive,
+  onOptionChange,
+}: EngineCardProps): JSX.Element => {
+  const logo = getEngineLogo(option.value);
+
+  const handleClick = useCallback(() => {
+    onOptionChange?.(option.value);
+  }, [option, onOptionChange]);
+
+  return (
+    <EngineCardRoot
+      role="option"
+      id={getListOptionId(rootId, option)}
+      isActive={isActive}
+      onClick={handleClick}
+    >
+      {logo ? (
+        <EngineCardImage src={logo} />
+      ) : (
+        <EngineCardIcon name="database" />
+      )}
+      <EngineCardTitle>{option.name}</EngineCardTitle>
+    </EngineCardRoot>
+  );
+};
+
+interface EngineEmptyStateProps {
+  isHosted?: boolean;
+}
+
+const EngineEmptyState = ({ isHosted }: EngineEmptyStateProps): JSX.Element => {
+  return (
+    <EngineEmptyStateRoot>
+      <EngineEmptyIcon name="search" size={32} />
+      {isHosted ? (
+        <EngineEmptyText>{t`Didn’t find anything`}</EngineEmptyText>
+      ) : (
+        <EngineEmptyText>{jt`Don’t see your database? Check out our ${(
+          <ExternalLink href={Settings.docsUrl("developers-guide-drivers")}>
+            {t`Community Drivers`}
+          </ExternalLink>
+        )} page to see if it’s available for self-hosting.`}</EngineEmptyText>
+      )}
+    </EngineEmptyStateRoot>
+  );
+};
+
+interface EngineToggleProps {
+  isExpanded: boolean;
+  onExpandedChange: (isExpanded: boolean) => void;
+}
+
+const EngineToggle = ({
+  isExpanded,
+  onExpandedChange,
+}: EngineToggleProps): JSX.Element => {
+  const handleClick = useCallback(() => {
+    onExpandedChange(!isExpanded);
+  }, [isExpanded, onExpandedChange]);
+
+  return (
+    <EngineToggleRoot
+      primary
+      iconRight={isExpanded ? "chevronup" : "chevrondown"}
+      onClick={handleClick}
+    >
+      {isExpanded ? t`Show less options` : t`Show more options`}
```

Comment:
"Show fewer options" makes more sense to me here because the options are countable